batchupload/uploader.py

```diff
@@ -19,7 +20,7 @@ def upload_single_file(file_name, media_file, text, target_site,
     Upload a single file in chunks.
 
     @param file_name: sanitized filename to use for upload
-    @param media_file: path to media file to upload
+    @param media_file: path or URL to media file to upload
     @param text: file description page
     @param target_site: pywikibot.Site object to which file should be uploaded
     @param chunk_size: Size of chunks (in MB) in which to upload file
```

Comment:
I'm not entirely familiar with chunking myself but I believe the idea is that you want large enough chunks to make the transfer efficient while being small enough to prevent the connection from having time to run into some hickup.

As such I'm not sure whether having the chunk size scale with the file size is desirable or not. It's definitely something we could investigate in a follow-up task though. A good first step would be to see how UploadWizard decides chunk size.